src/data/mod_json.rs

```diff
@@ -67,21 +67,51 @@ pub struct FileCopy {
     pub destination: String,
 }
 
+pub struct PreProcessingData {
+    pub version: String,
+    pub mod_id: String,
+}
+
 impl ModJson {
-    pub fn read() -> ModJson {
-        let mut file = std::fs::File::open("mod.json").expect("Opening mod.json failed");
+    pub fn get_template_name() -> &'static str {
+        "mod.template.json"
+    }
+
+    pub fn get_result_name() -> &'static str {
+        "mod.json"
+    }
+
+    pub fn read_and_preprocess(preprocess_data: &PreProcessingData, path: PathBuf) -> Self {
+        let mut file = std::fs::File::open(path).expect("Opening mod.json failed");
+
+        // Get data
         let mut json = String::new();
         file.read_to_string(&mut json).expect("Reading data failed");
 
-        serde_json::from_str::<ModJson>(&json).expect("Deserializing package failed")
+        // Pre process
+        let processsed = Self::preprocess(json, preprocess_data);
+
+        
+        serde_json::from_str(&processsed).expect("Deserializing package failed")
     }
 
-    pub fn write(&self) {
-        let json = serde_json::to_string_pretty(&self).expect("Serialization failed");
+    fn preprocess(s: String, preprocess_data: &PreProcessingData) -> String {
+        s
+        .replace("${version}", &preprocess_data.version)
+        .replace("${mod_id}", &preprocess_data.mod_id)
+    }
+
+    pub fn read(path: PathBuf) -> ModJson {
```

Comment:
isn't the mod json always in the root where the qpm json is as well, why require a path here?

Reply:
same goes for write ig